### ISSUE TITLE
The GraphCast generates different predictions with the same inputs. How to fix it?

### ISSUE THREAD

**Issue by LQscience** [google-deepmind/graphcast#123]

**Comment by LQscience** [comment]
I'm running GraphCast with the same inputs and random seed, but I'm getting inconsistent results. Could you please guide me on how to properly set the random seed so that I can obtain the same predictions under the same conditions?

Here's the random seed setting I'm using:
```python
import jax
import random
import numpy as np

seed = 42
random.seed(seed)
np.random.seed(seed)
jax.random.PRNGKey(seed)
```

**Comment by andrewlkd** [comment]
Hey, can you elaborate on what you've found/are trying? Some reproducer code would be useful.

**Comment by andrewlkd** [comment]
Could you confirm which hardware you're running on?

**Comment by mjwillson** [comment]
I assume you're running on GPU. GPUs are non-deterministic by default. You can set some flags which disable non-determinism but they will slow things down considerably.
Some further tips and discussion on this in the thread here: https://github.com/google-deepmind/graphcast/issues/79.

**Comment by LQscience** [comment]
> Could you confirm which hardware you're running on?

Thank you for the kind reply. The NVIDIA A100 was used in the test. I saw Matthew's reply and also checked out the thread in #79, which was very helpful. It resolved my question. Thanks again for your help!


> I assume you're running on GPU. GPUs are non-deterministic by default. You can set some flags which disable non-determinism but they will slow things down considerably. Some further tips and discussion on this in the thread here: [#79](https://github.com/google-deepmind/graphcast/issues/79).

Thank you for the clarification! As you mentioned, the non-deterministic behavior of GPUs seems to be the cause here. I’ve also gone through the discussion in #79, and it was very helpful—it resolved my question completely.

Thanks again for pointing me in the right direction!
